### DISCUSSION TITLE
I can't start flash on Pale Moon

### DISCUSSION THREAD

**Discussion by Gregglikehipi** [lightspark/lightspark#1072] [Q&A]
I did things from manual for instaling but flash doesnt work. I create plugins folder in Root Directory and Local directory. Put files in but flash didnt work. I tried to look for plugins and only saw flash plugin. May be the must be second plugin?

**Comment by dbluelle** [comment]
> I tried to look for plugins and only saw flash plugin.

Does that mean that you have an entry "Adobe Flash" in the addon manager under "plugins"?
If yes, then lightspark is correctly installed and should work.
What does it say if you click on "more" in the "Adobe Flash" entry?

Do you have an example of a page that doesn't work?

**Reply by Gregglikehipi** [reply]
What does it say if you click on "more" in the "Adobe Flash" entry?
The said that file is nplightsparkplugin.dll.
Do you have an example of a page that doesn't work?
On any site i try to run flash. On a place of flash appear black screen and then it become blank with no interaction:/

**Reply by dbluelle** [reply]
OK, so the plugin seems to be properly installed, good.
For some reason it doesn't seem to render correctly, so we have to dig deeper...
Please create an output log of Lightspark by setting the following environment variables:
LIGHTSPARK_PLUGIN_LOGLEVEL=4
LIGHTSPARK_PLUGIN_LOGFILE=<path_to_log_file> 
(with <path_to_log_file> pointing to a file in a writable folder)
then start Pale Moon and navigate to a site that doesn't work.
After that, post the created logfile here.
Please also post the url of the site you tested.

**Reply by Gregglikehipi** [reply]
Sorry for long waiting.
Where I can find "enviromental variables" file.
There link to site: https://www.maxgames.com/play/plazma-burst-2.html

**Reply by dbluelle** [reply]
environment variables are not set by a file.
Instead you set them in the terminal (cmd.exe on windows).
I'm not entirely sure but I think on windows it would be done by opening cmd.exe and then typing
```
set LIGHTSPARK_PLUGIN_LOGLEVEL=4
set LIGHTSPARK_PLUGIN_LOGFILE=<path_to_log_file>
```
and then start Pale Moon from that terminal:
```
<path_to_palemoon_installation>\palemoon.exe
```
